In [6]:
!pip install requests



In [7]:
import requests
import json


In [11]:
def wiki_search(search_term, limit=5):
    """
    Perform a Wikipedia API search and return results
    """
    url = "https://en.wikipedia.org/w/api.php"
    params = {
        "action": "query",
        "list": "search",
        "srsearch": search_term,
        "format": "json",
        "srlimit": limit
    }

    response = requests.get(url, params=params)
    return response.json()

def analyze_results(search_term):
    """
    Analyze and print search results
    """
    print(f"\nSearch Analysis for: {search_term}")
    print("-" * 50)

    results = wiki_search(search_term)

    for i, result in enumerate(results["query"]["search"], 1):
        title = result["title"]
        pageid = result["pageid"]
        snippet = result["snippet"]

        print(f"\n{i}. Title: {title}")
        print(f"   PageID: {pageid}")
        print(f"   URL: https://en.wikipedia.org/?curid={pageid}")
        print(f"   Snippet: {snippet[:100]}...")

# Perform three different searches
search_terms = ["Hyrax", "Pangolin", "Cat"]

for term in search_terms:
    analyze_results(term)




Search Analysis for: Hyrax
--------------------------------------------------

1. Title: Hyrax
   PageID: 28332898
   URL: https://en.wikipedia.org/?curid=28332898
   Snippet: Hyraxes (from Ancient Greek ὕραξ <span class="searchmatch">hýrax</span> 'shrew-mouse'), also called ...

2. Title: Rock hyrax
   PageID: 1239396
   URL: https://en.wikipedia.org/?curid=1239396
   Snippet: The rock <span class="searchmatch">hyrax</span> (/ˈhaɪ.ræks/; Procavia capensis), also called dassie...

3. Title: Tree hyrax
   PageID: 12541564
   URL: https://en.wikipedia.org/?curid=12541564
   Snippet: The tree <span class="searchmatch">hyrax</span> or tree dassie is a small nocturnal mammal native to...

4. Title: Yellow-spotted rock hyrax
   PageID: 12541741
   URL: https://en.wikipedia.org/?curid=12541741
   Snippet: The yellow-spotted rock <span class="searchmatch">hyrax</span> or bush <span class="searchmatch">hyr...

5. Title: Benin tree hyrax
   PageID: 67991924
   URL: https://en.wikipedia.org/?cur

# Wikipedia Search Results Comparison

## Hyrax Search
**API Results:**
1. Hyrax (PageID: 28332898)
2. Rock hyrax (PageID: 1239396)
3. Tree hyrax (PageID: 12541564)
4. Yellow-spotted rock hyrax (PageID: 12541741)
5. Benin tree hyrax (PageID: 67991924)

**GUI Results:**
1. Hyrax
2. Hyrax Hill
3. Hyrax Hill Prehistoric Site and Museum
4. Rock hyrax
5. Hyraxia Books

**Analysis:** The API focuses more on biological/species variants, while GUI includes geographical and cultural references (Hyrax Hill, Hyraxia Books).

## Pangolin Search
**API Results:**
1. Pangolin (PageID: 51124)
2. Phylogenetic Assignment Named Global Outbreak Lineages
3. Ground pangolin (PageID: 252144)
4. Pangolin (disambiguation)
5. Chinese pangolin (PageID: 10444089)

**GUI Results:**
1. Pangolin
2. Pangolin trade
3. Pangolin (disambiguation)
4. Pangolinisis
5. PANGOL (lineages)

**Analysis:** API results are more species-focused, while GUI includes more contextual topics like illegal trade. Both include the disambiguation page.

## Cat Search
**API Results:**
1. Cat (PageID: 6678)
2. Cat (disambiguation)
3. .cat
4. CatCat
5. Schrödinger's cat

**GUI Results:**
1. Cat
2. Catholic Church
3. Catherine the Great
4. Catherine, Princess of Wales
5. Catherine of Aragon

**Analysis:** API results focus more on direct cat-related entries and technical uses of "cat", while GUI results branch into broader terms starting with "cat" including historical figures and institutions.